src/plugins/intel_cpu/tests/functional/single_layer_tests/select.cpp

```diff
@@ -1,174 +1,204 @@
+public:
+   static std::string getTestCaseName(testing::TestParamInfo<selectParams> obj) {
+       std::vector<InputShape> shapes;
+       ElementType precision;
+       ngraph::op::AutoBroadcastSpec broadcast;
+       fusingSpecificParams fusingParams;
+       std::tie(shapes, precision, broadcast, fusingParams) = obj.param;
+
+       std::ostringstream result;
+       result << "Condition_prc_" << ElementType::boolean << "_Then_Else_prc_" << precision << "_";
+       result << "IS=(";
+       for (const auto& shape : shapes) {
+           result << shape.first << "_";
+       }
+       result << ")_TS=(";
+       for (const auto& shape : shapes) {
+           for (const auto& item : shape.second) {
+               result << CommonTestUtils::vec2str(item) << "_";
+           }
+       }
+       result << "Broadcast=" << broadcast.m_type;
+       result << CpuTestWithFusing::getTestCaseName(fusingParams);
+
+       return result.str();
+   }
+
+protected:
+   void SetUp() override {
+        abs_threshold = 0;
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+        std::vector<InputShape> shapes;
+        ElementType precision;
+        ngraph::op::AutoBroadcastSpec broadcast;
+        fusingSpecificParams fusingParams;
+        std::tie(shapes, precision, broadcast, fusingParams) = this->GetParam();
+        init_input_shapes(shapes);
+        std::tie(inFmts, outFmts, priority, selectedType) = emptyCPUSpec;
+        selectedType = makeSelectedTypeStr(getPrimitiveType(), ov::element::i8);
+
+        auto parameters = ngraph::builder::makeDynamicParams(ov::element::TypeVector{ov::element::boolean, precision, precision}, inputDynamicShapes);
+        auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(parameters));
+        auto select = ngraph::builder::makeSelect(paramOuts, broadcast);
+
+        function = makeNgraphFunction(precision, parameters, select, "Eltwise");
+   }
+
+   void generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) override {
+        inputs.clear();
+        const auto& modelInputs = function->inputs();
+        auto condTensor = ov::test::utils::create_and_fill_tensor(modelInputs[0].get_element_type(), targetInputStaticShapes[0], 3, -1, 2);
+        auto thenTensor = ov::test::utils::create_and_fill_tensor(modelInputs[1].get_element_type(), targetInputStaticShapes[1], 10, -10, 2);
+        auto elseTensor = ov::test::utils::create_and_fill_tensor(modelInputs[2].get_element_type(), targetInputStaticShapes[2], 10, 0, 2);
+        inputs.insert({modelInputs[0].get_node_shared_ptr(), condTensor});
+        inputs.insert({modelInputs[1].get_node_shared_ptr(), thenTensor});
+        inputs.insert({modelInputs[2].get_node_shared_ptr(), elseTensor});
+    }
+};
+
+TEST_P(SelectLayerCPUTest, CompareWithRefs) {
+    run();
+    CheckPluginRelatedResults(compiledModel, "Eltwise");
+}
+
+const std::vector<ElementType> precisions = {
```

Comment:
Das it make sense to add `i8` precision as well?

Reply:
Done, thanks